File: lab2/catkin_ws/src/lab2_part1_pkg/src/my_first_node.py
#!/usr/bin/env python
import rospy

if __name__ == "__main__":
    rospy.init_node("hello_world_py_node")
    log_freq = rospy.get_param('/hello_publish_frequency', 1)
    rate = rospy.Rate(log_freq)

    while not rospy.is_shutdown():
        rospy.loginfo("Hello World!")
        rate.sleep()

    rospy.loginfo("Node is shutting down...")

    

File: lab2/catkin_ws/src/lab2_part1_pkg/src/my_first_node.cpp
#include <ros/ros.h>

int main(int argc, char **argv)
{
    ros::init(argc, argv, "hello_word_cpp_node");
    ros::NodeHandle nh;

    double pub_freq;
    nh.param("/hello_publish_frequency", pub_freq, 1.0);
    ros::Rate rate(pub_freq);

    while(ros::ok())
    {
        ROS_INFO("Hello World!");
        rate.sleep();
    }

    return 0;
}